From 72263b45e4e9bab3179cd69bbc0a58bf5c91764d Mon Sep 17 00:00:00 2001
From: Vincent Driessen <vincent@3rdcloud.com>
Date: Thu, 19 Dec 2013 16:24:39 +0100
Subject: [PATCH] Sanitize the Python 2 check.

This broke for me on a project that has a module called 'builtins'.
Let's just do a proper check using the normal `sys.version_info` API.
---
 pyflakes/checker.py | 11 ++++++-----
 1 file changed, 6 insertions(+), 5 deletions(-)

diff --git a/pyflakes/checker.py b/pyflakes/checker.py
index d59b8c9..4f65e46 100644
--- a/pyflakes/checker.py
+++ b/pyflakes/checker.py
@@ -7,12 +7,13 @@
 import doctest
 import os
 import sys
-try:
-    builtin_vars = dir(__import__('builtins'))
-    PY2 = False
-except ImportError:
-    builtin_vars = dir(__import__('__builtin__'))
+
+if sys.version_info < (3, 0):
     PY2 = True
+    builtin_vars = dir(__import__('__builtin__'))
+else:
+    PY2 = False
+    builtin_vars = dir(__import__('builtins'))
 
 try:
     import ast
-- 
1.8.5.1

